src/etc/inc/system.inc

```diff
@@ -1049,14 +1044,19 @@ EOD;
 	if (isset($config['installedpackages']['package'])) {
 		foreach ($config['installedpackages']['package'] as $package) {
 			if (isset($package['logging']['logsocket']) && $package['logging']['logsocket'] != '' &&
-			    is_dir(dirname($package['logging']['logsocket'])) &&
 			    !in_array($package['logging']['logsocket'], $log_sockets)) {
 				$log_sockets[] = $package['logging']['logsocket'];
 			}
 		}
 	}
+	
 	$syslogd_sockets = "";
 	foreach ($log_sockets as $log_socket) {
+		// Ensure that the log directory exists
+		$logpath = dirname($log_socket);
+		if (!is_dir($logpath)) {
+			mkdir($logpath, 0777, true);
```

Comment:
I put it because if not specified it would be the default value for mkdir. But to make sure if intermediate paths are missing they will be created i thought to specify the recursive flag.. Its just that mode parameter sits in between..
`bool mkdir ( string $pathname [, int $mode = 0777 [, bool $recursive = false [, resource $context ]]] )`

However i will switch to safe_mkdir() which will do it all at once..